### PR TITLE
Envoy config generator from swagger

### PR DESCRIPTION
**Pull request by chashikajw**
### Purpose

As $subject, this function will generates the envoy artifacts which are listener, routes, clusters and endpoints for xds apis using swagger files.  
Multiple swagger files are supported.

PS:

**Endpoint format:**

[protocol(optional)][host][port][basepath]
Ex1: https://petstore.swagger.io:8080/api/v3
Ex2:[ petstore.swagger.io:8080/api/v3](url)

In the swagger files, endpoints should be included in the following ways. 

**Priority 1:**

x-wso2-production-endpoints:
  urls:
    - 'https://petstore.swagger.io/api/v3'
  type: https
x-wso2-sandbox-endpoints:
  urls:
    - 'http://example.com:8080/api/v3'
  type: http

**OpenAPI 3:**
servers:
  - url: 'https://petstore.swagger.io:8080/api/v3'

**Swagger2**

host: "petstore.swagger.io:8080"
basePath: "/api/v3"

### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

**Comment by menakaj** [comment]
Related issue : https://github.com/wso2/product-microgateway/issues/1238

File: internal/pkg/oasparser/config/config.go [hunk 1]
```diff
@@ -16,9 +16,12 @@
...
 const (
-	LISTENER_ADDRESS   string = "0.0.0.0"
-	LISTENER_PORT      uint32 = 10000
-	API_PORT           uint32 = 8080
-	MANAGER_STATPREFIX string = "ingress_http"
+	LISTENER_ADDRESS  string = "0.0.0.0"
+	LISTENER_PORT     uint32 = 10000
+	API_DEFAULT_PORT   uint32 = 8080
+	MANAGER_STATPREFIX    string = "ingress_http"
+	CLUSTER_CONNECT_TIMEOUT    = 5 * time.Second
```
Comment (dinusha92) [inline internal/pkg/oasparser/config/config.go:26]: Formatting issue. Shall we fix this?

File: internal/pkg/oasparser/constants/openApiConstants.go [hunk 2]
```diff
@@ -16,5 +16,8 @@
  */
 package constants
 
-const PRODUCTION_ENDPOINTS string = "x-wso2-production-endpoints"
-const SANDBOX_ENDPOINTS string = "x-wso2-sandbox-endpoints"
+const (
+	PRODUCTION_ENDPOINTS string = "x-wso2-production-endpoints"
+	SANDBOX_ENDPOINTS string = "x-wso2-sandbox-endpoints"
+)
```
Comment (dinusha92) [inline internal/pkg/oasparser/constants/openApiConstants.go:22]: Formatting issue.

File: internal/pkg/oasparser/envoyCodegen/routesWithClusters.go [hunk 3]
```diff
@@ -173,47 +166,104 @@ func createCluster(address core.Address, clusterName string) v2.Cluster {
...
+						EngineType: &envoy_type_matcher.RegexMatcher_GoogleRe2{
+							GoogleRe2: &envoy_type_matcher.RegexMatcher_GoogleRE2{
+								MaxProgramSize: nil,
+							},
 						},
+						Regex: routepath,
 					},
-					Regex: resourcePath,
 				},
+				/*Headers: []*v2route.HeaderMatcher {
```
Comment (dinusha92) [inline internal/pkg/oasparser/envoyCodegen/routesWithClusters.go:204]: What is the purpose of the commented code. If there's a reason shall we mention that in a comment? If not, let's remove this :)

File: internal/pkg/oasparser/models/apiDefinition/mgwResource.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package apiDefinition
```
Comment (dinusha92) [inline internal/pkg/oasparser/models/apiDefinition/mgwResource.go:1]: License header missing.

File: internal/pkg/oasparser/config/config.go (hunk 1 above)
Comment (chashikajw) [reply internal/pkg/oasparser/config/config.go:26]: This is changed in the repo

File: internal/pkg/oasparser/constants/openApiConstants.go (hunk 2 above)
Comment (chashikajw) [reply internal/pkg/oasparser/constants/openApiConstants.go:22]: This is already fixed

File: internal/pkg/oasparser/envoyCodegen/routesWithClusters.go (hunk 3 above)
Comment (chashikajw) [reply internal/pkg/oasparser/envoyCodegen/routesWithClusters.go:204]: This is already removed

File: internal/pkg/oasparser/models/apiDefinition/mgwResource.go (hunk 4 above)
Comment (chashikajw) [reply internal/pkg/oasparser/models/apiDefinition/mgwResource.go:1]: This is already added

**Pull request merged**